python/mxnet/contrib/quantization.py

```diff
@@ -139,18 +136,20 @@ def __init__(self, include_layer=None, logger=None):
 
     def collect(self, name, arr):
         """Callback function for collecting layer output NDArrays."""
-        name = py_str(name)
-        if self.include_layer is not None and not self.include_layer(name):
-            return
-        handle = ctypes.cast(arr, NDArrayHandle)
-        arr = NDArray(handle, writable=False).copyto(cpu())
-        if self.logger is not None:
-            self.logger.info("Collecting layer %s output of shape %s" % (name, arr.shape))
-        if name in self.nd_dict:
-            self.nd_dict[name].append(arr)
-        else:
-            self.nd_dict[name] = [arr]
-
+        try:
+            name = py_str(name)
+            if self.include_layer is not None and not self.include_layer(name):
+                return
+            handle = ctypes.cast(arr, NDArrayHandle)
+            arr = NDArray(handle, writable=False).copyto(cpu())
+            if self.logger is not None:
+                self.logger.info("Collecting layer %s output of shape %s" % (name, arr.shape))
+            if name in self.nd_dict:
+                self.nd_dict[name].append(arr)
+            else:
+                self.nd_dict[name] = [arr]
+        except KeyboardInterrupt:
```

Comment:
We should probably never explicitly cause the process to exit in a library like MXNet.  This could make it tricky to debug issues for applications that consume MXNet.  If you consume a library you don't want that library killing your process before you have a change to handle an error or log a failure.

Reply:
Adding this is to improve user friendly. Because without this change, you can't ctrl + c to terminate the quantizaiton script. @KellenSunderland For this case, do you have any suggested change?

Reply:
Yes just replace the sys exit with a throw.  This way if a service is calling the script and for whatever reason the exception gets triggered the service will have a chance to log the reason it's shutting down.

Reply:
> replace the sys exit with a throw.
This doesn't work because it is a c++ callback function written in python. You can't directly throw c++ exception from python. And also raise python exception doesn't help here as the c++ caller can't catch it.